cli/azd/pkg/auth/cache.go

```diff
@@ -52,6 +93,34 @@ func (a *msalCacheAdapter) Export(ctx context.Context, cache cache.Marshaler, _
 	return a.cache.Set(cCurrentUserCacheKey, val)
 }
 
+// Normalize keys by removing upper-case keys and replacing them with lower-case keys.
+// In the case where a lower-case key and upper-case key exists, the lower-case key entry
+// takes precedence.
+func normalizeKeys(m map[string]json.RawMessage) {
+	for k, v := range m {
+		if hasUpper(k) {
+			// An upper-case key entry exists. Delete it as it is no longer allowed.
+			delete(m, k)
+
+			// If a lower-case key entry exists, that supersedes it and we are done.
+			// Otherwise, we can safely upgrade the cache entry by re-adding it with lower case.
+			lower := strings.ToLower(k)
+			if _, isLower := m[lower]; !isLower {
+				m[lower] = v
+			}
+		}
+	}
+}
+
+func hasUpper(s string) bool {
```

Comment:
I think we don't really need this

Reply:
Kept as-is based on https://github.com/Azure/azure-dev/pull/2664#discussion_r1309353591